### ISSUE TITLE
mail_transfer_agent issue

### ISSUE THREAD

**Issue by aaronlippold** [USGS-CIDA/stig#109]
hi all I am getting this? Has anyone seen this error on RHEL 7.3/7.4 before?

```
 Recipe: stig::mail_transfer_agent
         * template[/etc/postfix/main.cf] action create

           ================================================================================
           Error executing action `create` on resource 'template[/etc/postfix/main.cf]'
           ================================================================================

           Chef::Exceptions::FileNotFound
           ------------------------------
           Cookbook 'stig' (0.6.11) does not contain a file at any of these locations:
             templates/redhat-7.4/
             templates/redhat/
             templates/default/
             templates/

           This cookbook _does_ contain: ['templates/default/etc_grubd_40_custom.erb','templates/default/etc_audit_auditd.conf.erb','templates/default/audit_rules.erb','templates/default/etc_hosts.allow.erb','templates/default/aide.conf.erb','templates/default/etc_hosts.deny.erb','templates/default/etc_modprobe.d_ipv6.conf.erb','templates/default/etc_login.defs.erb','templates/default/etc_init_isc-dhcp-server6.conf.erb','templates/default/etc_init_isc-dhcp-server.conf.erb','templates/default/etc_pam.d_common-password.erb','templates/default/etc_main.cf_rhel.erb','templates/default/etc_selinux_config.erb','templates/default/etc_pam_d_password_system_auth.erb','templates/default/etc_rsyslog.conf.erb','templates/default/etc_sysctl.conf_ubuntu.erb','templates/default/etc_modprobe.d_CIS.conf.erb','templates/default/etc_sysctl.conf_rhel.erb','templates/default/etc_main.cf_ubuntu.erb','templates/default/etc_ssh_sshd_config.erb','templates/default/etc_sysconfig_network.erb','templates/default/issue.erb','templates/default/etc_sysconfig_init.erb','templates/default/limits.conf.erb','templates/default/issue.net.erb','templates/default/selinux_enforce.erb','templates/default/motd.erb']

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/stig/recipes/mail_transfer_agent.rb

            21: template '/etc/postfix/main.cf' do
            22:   source source
            23:   owner 'root'
            24:   group 'root'
            25:   mode 0o644
            26:   notifies :restart, 'service[postfix]', :immediately
            27: end
            28:

           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/stig/recipes/mail_transfer_agent.rb:21:in `from_file'

           template("/etc/postfix/main.cf") do
             action [:create]
             default_guard_interpreter :default
             declared_type :template
             cookbook_name "stig"
             recipe_name "mail_transfer_agent"
             owner "root"
             group "root"
             mode 420
             path "/etc/postfix/main.cf"
             verifications []
           end

           System Info:
           ------------
           chef_version=13.7.16
           platform=redhat
           platform_version=7.4
           ruby=ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-linux]
           program_name=chef-solo worker: ppid=19931;start=21:37:22;
           executable=/opt/chef/bin/chef-solo

       Recipe: stig::boot_settings
         * execute[toggle_selinux] action run (skipped due to not_if)
```

**Comment by isuftin** [comment]
@aaronlippold I'm unable to reproduce this. Try to destroy your current VM and re-run Test Kitchen.

Here's my local Kitchen YAML I'm testing with:
```
---
driver:
  name: vagrant

provisioner:
  name: chef_zero

verifier:
  name: inspec

platforms:
  - name: centos-6
  - name: centos-7

suites:
  - name: centos7
    excludes:
      - centos-6
    verifier:
      inspec_tests:
          - name: centos7
            path: test/integration/centos7/inspec
    run_list:
      - recipe[stig::mail_transfer_agent]
    attributes: {
      "auditd" : {
        "ruleset" : "stig"
      }
    }
```

Output logs:
```
$ bundle exec kitchen converge centos7-centos-7                                                                       [394/20934]
-----> Starting Kitchen (v1.20.0)
-----> Creating <centos7-centos-7>...
       Bringing machine 'default' up with 'virtualbox' provider...
       ==> default: Importing base box 'bento/centos-7'...
==> default: Matching MAC address for NAT networking...
       ==> default: Checking if box 'bento/centos-7' is up to date...
       ==> default: Setting the name of the VM: centos7-centos-7_default_1519221812528_54515
       ==> default: Clearing any previously set network interfaces...
       ==> default: Preparing network interfaces based on configuration...
           default: Adapter 1: nat
       ==> default: Forwarding ports...
           default: 22 (guest) => 2222 (host) (adapter 1)
       ==> default: Booting VM...
       ==> default: Waiting for machine to boot. This may take a few minutes...
           default: SSH address: 127.0.0.1:2222
           default: SSH username: vagrant
           default: SSH auth method: private key
           default:
           default: Vagrant insecure key detected. Vagrant will automatically replace
           default: this with a newly generated keypair for better security.
           default:
           default: Inserting generated public key within guest...
           default: Removing insecure key from the guest if it's present...
           default: Key inserted! Disconnecting and reconnecting using new SSH key...
       ==> default: Machine booted and ready!
       ==> default: Checking for guest additions in VM...
       ==> default: Setting hostname...
       ==> default: Mounting shared folders...
           default: /tmp/omnibus/cache => /Users/isuftin/.kitchen/cache
       ==> default: Uploading root certificates to guest instance...
       ==> default: -- /Users/isuftin/.chef/trusted_certs/root.cer => /etc/pki/ca-trust/source/anchors/root.cer
       ==> default: [vagrant-hostmanager:guests] Updating hosts file on active guest virtual machines...
       ==> default: [vagrant-hostmanager:host] Updating hosts file on your workstation (password may be required)...
       ==> default: Machine not provisioned because `--no-provision` is specified.
       [SSH] Established
       Vagrant instance <centos7-centos-7> created.
       Finished creating <centos7-centos-7> (0m39.29s).
-----> Converging <centos7-centos-7>...
       Preparing files for transfer
       Preparing dna.json
       Resolving cookbook dependencies with Berkshelf 6.3.1...
       Removing non-cookbook files before transfer
       Preparing validation.pem
       Preparing client.rb
-----> Installing Chef Omnibus (install only if missing)
       Downloading https://omnitruck.chef.io/install.sh to file /tmp/install.sh
       Trying wget...
       Download complete.
       el 7 x86_64
       Getting information for chef stable  for el...
       downloading https://omnitruck.chef.io/stable/chef/metadata?v=&p=el&pv=7&m=x86_64
         to file /tmp/install.sh.3698/metadata.txt
       trying wget...
       sha1     0d8b0f9b3ce0a2721f7e8ff7ea46330fd712df46
       sha256   89c96addb749370044964b63d16382149f0756d54e833602dd91cf039a92a9f7
       url      https://packages.chef.io/files/stable/chef/13.7.16/el/7/chef-13.7.16-1.el7.x86_64.rpm
       version  13.7.16
       downloaded metadata file looks valid...
       /tmp/omnibus/cache/chef-13.7.16-1.el7.x86_64.rpm exists
       Comparing checksum with sha256sum...

       WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING

       You are installing an omnibus package without a version pin.  If you are installing
       on production servers via an automated process this is DANGEROUS and you will
       be upgraded without warning on new releases, even to new major releases.
       Letting the version float is only appropriate in desktop, test, development or
       CI/CD environments.

       WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING

       Installing chef
       installing with rpm...
       warning: /tmp/omnibus/cache/chef-13.7.16-1.el7.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY
       Preparing...                          ################################# [100%]
       Updating / installing...
          1:chef-13.7.16-1.el7               ################################# [100%]
       Thank you for installing Chef!
       Transferring files to <centos7-centos-7>
       Starting Chef Client, version 13.7.16
       Creating a new client identity for centos7-centos-7 using the validator key.
       resolving cookbooks for run list: ["stig::mail_transfer_agent"]
       Synchronizing Cookbooks:
         - logrotate (2.2.0)
         - sysctl (0.10.2)
         - ohai (5.2.1)
         - stig (0.6.11)
       Installing Cookbook Gems:
       Compiling Cookbooks...
       Converging 2 resources
       Recipe: stig::mail_transfer_agent
         * template[/etc/postfix/main.cf] action create
           - update content in file /etc/postfix/main.cf from 0bd87e to d1b62e
           --- /etc/postfix/main.cf     2014-06-10 01:39:24.000000000 +0000
           +++ /etc/postfix/.chef-main20180221-3827-b3nwsd.cf   2018-02-21 14:04:27.524096582 +0000
           @@ -1,3 +1,9 @@
           +#
           +# Generated by Chef for centos7-centos-7.vagrantup.com.
           +# Local modifications will be overwritten.
           +#
           +# See https://github.com/USGS-CIDA/stig/blob/master/templates/default/main.cf.erb
           +
            # Global Postfix configuration file. This file lists only a subset
            # of all parameters. For the syntax, and for a complete parameter
            # list, see the postconf(5) manual page (command: "man 5 postconf").
           @@ -20,7 +26,9 @@
            # is no cure for address rewriting mistakes or mail routing mistakes.
            #
            #soft_bounce = no
           +soft_bounce = no

           +
            # LOCAL PATHNAME INFORMATION
            #
            # The queue_directory specifies the location of the Postfix queue.
           @@ -63,10 +71,10 @@
            # These rights are used in the absence of a recipient user context.
            # DO NOT SPECIFY A PRIVILEGED USER OR THE POSTFIX OWNER.
            #
           -#default_privs = nobody
           +default_privs = nobody

            # INTERNET HOST AND DOMAIN NAMES
           -#
           +#
            # The myhostname parameter specifies the internet hostname of this
            # mail system. The default is to use the fully-qualified domain name
            # from gethostname(). $myhostname is used as a default value for many
           @@ -83,7 +91,7 @@
            #mydomain = domain.tld

            # SENDING MAIL
           -#
           +#
            # The myorigin parameter specifies the domain that locally-posted
            # mail appears to come from. The default is to append $myhostname,
            # which is fine for small sites.  If you run a domain with multiple
           @@ -97,7 +105,6 @@
            #
            #myorigin = $myhostname
            #myorigin = $mydomain
           -
            # RECEIVING MAIL

            # The inet_interfaces parameter specifies the network interface
           @@ -115,8 +122,9 @@
            #inet_interfaces = $myhostname, localhost
            inet_interfaces = localhost

           +
            # Enable IPv4, and IPv6 if supported
           -inet_protocols = all
           +inet_protocols = ipv4

            # The proxy_interfaces parameter specifies the network interface
            # addresses that this mail system receives mail on by way of a
           @@ -161,10 +169,9 @@
            #
            # See also below, section "REJECTING MAIL FOR UNKNOWN LOCAL USERS".
            #
           -mydestination = $myhostname, localhost.$mydomain, localhost
            #mydestination = $myhostname, localhost.$mydomain, localhost, $mydomain
           -#mydestination = $myhostname, localhost.$mydomain, localhost, $mydomain,
           -#   mail.$mydomain, www.$mydomain, ftp.$mydomain
           +#mydestination = $myhostname, localhost.$mydomain, localhost, $mydomain, mail.$mydomain, www.$mydomain, ftp.$mydomain
           +mydestination = $myhostname, localhost.$mydomain, localhost

            # REJECTING MAIL FOR UNKNOWN LOCAL USERS
            #
           @@ -184,7 +191,7 @@
            #
            # - You define $mydestination domain recipients in files other than
            #   /etc/passwd, /etc/aliases, or the $virtual_alias_maps files.
           -#   For example, you define $mydestination domain recipients in
           +#   For example, you define $mydestination domain recipients in
            #   the $virtual_mailbox_maps files.
            #
            # - You redefine the local delivery agent in master.cf.
           @@ -204,10 +211,11 @@
            # The right-hand side of the lookup tables is conveniently ignored.
            # In the left-hand side, specify a bare username, an @domain.tld
            # wild-card, or specify a user@domain.tld address.
           -#
           +#
            #local_recipient_maps = unix:passwd.byname $alias_maps
            #local_recipient_maps = proxy:unix:passwd.byname $alias_maps
            #local_recipient_maps =
           +#local_recipient_maps =

            # The unknown_local_recipient_reject_code specifies the SMTP server
            # response code when a recipient domain matches $mydestination or
           @@ -236,16 +244,16 @@
            # clients in the same IP subnetworks as the local machine.
            # On Linux, this does works correctly only with interfaces specified
            # with the "ifconfig" command.
           -#
           +#
            # Specify "mynetworks_style = class" when Postfix should "trust" SMTP
            # clients in the same IP class A/B/C networks as the local machine.
            # Don't do this with a dialup site - it would cause Postfix to "trust"
            # your entire provider's network.  Instead, specify an explicit
            # mynetworks list by hand, as described below.
           -#
           +#
            # Specify "mynetworks_style = host" when Postfix should "trust"
            # only the local machine.
           -#
           +#
            #mynetworks_style = class
            #mynetworks_style = subnet
            #mynetworks_style = host
           @@ -274,7 +282,7 @@
            # - from "untrusted" clients to destinations that match $relay_domains or
            #   subdomains thereof, except addresses with sender-specified routing.
            # The default relay_domains value is $mydestination.
           -#
           +#
            # In addition to the above, the Postfix SMTP server by default accepts mail
            # that Postfix is final destination for:
            # - destinations that match $inet_interfaces or $proxy_interfaces,
           @@ -282,7 +290,7 @@
            # - destinations that match $virtual_alias_domains,
            # - destinations that match $virtual_mailbox_domains.
            # These destinations do not need to be listed in $relay_domains.
           -#
           +#
            # Specify a list of hosts or domains, /file/name patterns or type:name
            # lookup tables, separated by commas and/or whitespace.  Continue
            # long lines by starting the next line with whitespace. A file name
           @@ -327,7 +335,7 @@
            # The right-hand side of the lookup tables is conveniently ignored.
            # In the left-hand side, specify an @domain.tld wild-card, or specify
            # a user@domain.tld address.
           -#
           +#
            #relay_recipient_maps = hash:/etc/postfix/relay_recipients

            # INPUT RATE CONTROL
           @@ -336,15 +344,15 @@
            # flow control. This feature is turned on by default, although it
            # still needs further development (it's disabled on SCO UNIX due
            # to an SCO bug).
           -#
           +#
            # A Postfix process will pause for $in_flow_delay seconds before
            # accepting a new message, when the message arrival rate exceeds the
            # message delivery rate. With the default 100 SMTP server process
            # limit, this limits the mail inflow to 100 messages a second more
            # than the number of messages delivered per second.
           -#
           +#
            # Specify 0 to disable the feature. Valid delays are 0..10.
           -#
           +#
            #in_flow_delay = 1s

            # ADDRESS REWRITING
           @@ -374,7 +382,7 @@
            # On systems with NIS, the default is to search the local alias
            # database, then the NIS alias database. See aliases(5) for syntax
           -#
           +#
            # If you change the alias database, run "postalias /etc/aliases" (or
            # wherever your system stores the mail alias file), or simply run
            # "newaliases" to build the necessary DBM or DB file.
           @@ -383,9 +391,9 @@
            # "postfix reload" to eliminate the delay.
            #
            #alias_maps = dbm:/etc/aliases
           -alias_maps = hash:/etc/aliases
            #alias_maps = hash:/etc/aliases, nis:mail.aliases
            #alias_maps = netinfo:/aliases
           +alias_maps = hash:/etc/aliases

            # The alias_database parameter specifies the alias database(s) that
            # are built with "newaliases" or "sendmail -bi".  This is a separate
           @@ -393,9 +401,8 @@
            # tables that are not necessarily all under control by Postfix.
            #
            #alias_database = dbm:/etc/aliases
           -#alias_database = dbm:/etc/mail/aliases
           -alias_database = hash:/etc/aliases
            #alias_database = hash:/etc/aliases, hash:/opt/majordomo/aliases
           +alias_database = hash:/etc/aliases

            # ADDRESS EXTENSIONS (e.g., user+foo)
            #
           @@ -417,7 +424,7 @@
            #
            #home_mailbox = Mailbox
            #home_mailbox = Maildir/
           -
           +
            # The mail_spool_directory parameter specifies the directory where
            # UNIX-style mailboxes are kept. The default setting depends on the
            # system type.
           @@ -459,11 +466,9 @@
            #
            # NOTE: if you use this feature for accounts not in the UNIX password
            # file, then you must update the "local_recipient_maps" setting in
           -# the main.cf file, otherwise the SMTP server will reject mail for
           +# the main.cf file, otherwise the SMTP server will reject mail for
            # non-UNIX accounts with "User unknown in local recipient table".
            #
           -# Cyrus IMAP over LMTP. Specify ``lmtpunix      cmd="lmtpd"
           -# listen="/var/imap/socket/lmtp" prefork=0'' in cyrus.conf.
            #mailbox_transport = lmtp:unix:/var/lib/imap/socket/lmtp

            # If using the cyrus-imapd IMAP server deliver local mail to the IMAP
           @@ -484,10 +489,9 @@
            # can be used to take advantage of the single instance message store
            # capability of Cyrus. The concurrency limit can be used to control
            # how many simultaneous LMTP sessions will be permitted to the Cyrus
           -# message store.
           +# message store.
            #
           -# Cyrus IMAP via command line. Uncomment the "cyrus...pipe" and
           -# subsequent line in master.cf.
           +# To use the old cyrus deliver program you have to set:
            #mailbox_transport = cyrus

            # The fallback_transport specifies the optional transport in master.cf
           @@ -501,7 +505,7 @@
            #
            # NOTE: if you use this feature for accounts not in the UNIX password
            # file, then you must update the "local_recipient_maps" setting in
           -# the main.cf file, otherwise the SMTP server will reject mail for
           +# the main.cf file, otherwise the SMTP server will reject mail for
            # non-UNIX accounts with "User unknown in local recipient table".
            #
            #fallback_transport = lmtp:unix:/var/lib/imap/socket/lmtp
           @@ -523,15 +527,15 @@
            #
            # NOTE: if you use this feature for accounts not in the UNIX password
            # file, then you must specify "local_recipient_maps =" (i.e. empty) in
           -# the main.cf file, otherwise the SMTP server will reject mail for
           +# the main.cf file, otherwise the SMTP server will reject mail for
            # non-UNIX accounts with "User unknown in local recipient table".
            #
            #luser_relay = $user@other.host
            #luser_relay = $local@other.host
            #luser_relay = admin+$local
           -
           +
            # JUNK MAIL CONTROLS
           -#
           +#
            # The controls listed here are only a very small subset. The file
            # SMTPD_ACCESS_README provides an overview.

           @@ -553,11 +557,11 @@
            # deferred mail, so that mail can be flushed quickly with the SMTP
            # "ETRN domain.tld" command, or by executing "sendmail -qRdomain.tld".
            # See the ETRN_README document for a detailed description.
           -#
           +#
            # The fast_flush_domains parameter controls what destinations are
            # eligible for this service. By default, they are all domains that
            # this server is willing to relay mail to.
           -#
           +#
            #fast_flush_domains = $relay_domains
            # SHOW SOFTWARE VERSION OR NOT
           @@ -581,14 +585,11 @@
            # too many are run at the same time. With SMTP deliveries, 10
            # simultaneous connections to the same domain could be sufficient to
            # raise eyebrows.
           -#
           +#
            # Each message delivery transport has its XXX_destination_concurrency_limit
            # parameter.  The default is $default_destination_concurrency_limit for
            # most delivery transports. For the local delivery agent the default is 2.

           -#local_destination_concurrency_limit = 2
           -#default_destination_concurrency_limit = 20
           -
            # DEBUGGING CONTROL
            #
            # The debug_peer_level parameter specifies the increment in verbose
           @@ -613,10 +614,6 @@
            # the process marches on. If you use an X-based debugger, be sure to
            # set up your XAUTHORITY environment variable before starting Postfix.
            #
           -debugger_command =
           -     PATH=/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin
           -     ddd $daemon_directory/$process_name $process_id & sleep 5
           -
            # If you can't use X, use this to capture the call stack when a
            # daemon crashes. The result is in a file in the configuration
            # directory, and is named after the process name and the process ID.
           @@ -635,14 +632,19 @@
            #   PATH=/bin:/usr/bin:/sbin:/usr/sbin; export PATH; screen
            #   -dmS $process_name gdb $daemon_directory/$process_name
            #   $process_id & sleep 1
           +debugger_command =
           +    PATH=/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin
           +    ddd $daemon_directory/$process_name $process_id & sleep 5

           +
           +
            # INSTALL-TIME CONFIGURATION INFORMATION
            #
            # The following parameters are used when installing a new Postfix version.
           -#
           +#
            # sendmail_path: The full pathname of the Postfix sendmail command.
            # This is the Sendmail-compatible mail posting interface.
           -#
           +#
            sendmail_path = /usr/sbin/sendmail.postfix

            # newaliases_path: The full pathname of the Postfix newaliases command.
           @@ -652,7 +654,7 @@

            # mailq_path: The full pathname of the Postfix mailq command.  This
            # is the Sendmail-compatible mail queue listing command.
           -#
           +#
            mailq_path = /usr/bin/mailq.postfix

            # setgid_group: The group for mail submission and queue management
           @@ -672,9 +674,9 @@
            # sample_directory: The location of the Postfix sample configuration files.
            # This parameter is obsolete as of Postfix 2.1.
            #
           -sample_directory = /usr/share/doc/postfix-2.10.1/samples
           +sample_directory = /usr/share/doc/postfix-2.6.6/samples

            # readme_directory: The location of the Postfix README files.
            #
           -readme_directory = /usr/share/doc/postfix-2.10.1/README_FILES
           +readme_directory =  /usr/share/doc/postfix-2.6.6/README_FILES
           - restore selinux security context
         * service[postfix] action restart
           - restart service service[postfix]
         * service[postfix] action nothing (skipped due to action :nothing)

       Running handlers:
       Running handlers complete
       Chef Client finished, 2/3 resources updated in 04 seconds
       Downloading files from <centos7-centos-7>
       Finished converging <centos7-centos-7> (0m28.42s).
-----> Kitchen is finished. (1m9.61s)
```

**Comment by isuftin** [comment]
@aaronlippold I notice you are working with RHEL specifically? If so, could you post your kitchen yaml so that I may try to reproduce with your settings?

**Comment by cpoma** [comment]
Error comes from -  the check of platform versus platform_family on RH.

if %w[rhel fedora centos].include?(node['platform'])
  source = 'etc_main.cf_rhel.erb'
end

Correct code should be:

# Fixed to check platform_family versus platform
#    'redhat', 'fedora', 'centos' are platforms;
#    'rhel' is the platform_family that includes those platforms
source = 'etc_main.cf_rhel.erb' if node['platform_family'] == 'rhel'

**Comment by isuftin** [comment]
@aaronlippold Feel free to close this if you feel it's been resolved

**Comment by aaronlippold** [comment]
Great work @cpoma, also we should sync up with @samcornwell as well today.
